src/pytesmo/validation_framework/validation.py

```diff
@@ -220,10 +223,24 @@ def calc(
         only_with_reference : bool, optional
             If this is enabled, only combinations that include the reference
             dataset (from the data manager) are calculated.
-        handle_errors: Literal['raise', 'ignore'], optional (default: 'raise')
-            `raise`: If an error occurs during validation, raise exception.
-            `ignore`: If an error occurs during validation, log it and
-             continue with next GPI.
+        handle_errors: str, optional (default: 'raise')
+            Governs how to handle errors::
+
+            * `raise`: If an error occurs during validation, raise exception.
+            * `returncode`: If an error occurs, assign the correct return code
+              to the result template and continue with the next GPI. The main
+              difference to `ignore` is that the result for this GPI will still
+              be returned, but the metric values will be set to NaN. `ignore`
+              just completely skips over the GPI and does not return a result
+              for it.
+            * `ignore`: If an error occurs during validation, log it and
+               continue with next GPI.
+            * `deprecated`: Mix of `raise` and `ignore` to reproduce the
```

Comment:
The idea here was to have to options that try to reproduce the old error handling as closely as possible:
- "deprecated" reproduces the old "raise" option, where some of the errors where raised, but some of the things where I now raise an error were just ignored and an empty dictionary was returned as validation result, e.g. in case there was no data returned from the temporal matching.
- "ignore" reproduces the old "ignore" option, and in case of an error returns an empty dictionary as validation result.

The new options would now be:
- "raise": Always raise an error from an exception
- "returncode": like ignore, but instead of an empty dictionary, a validation result template with correct return codes and NaNs as metrics is returned.

I personally would also prefer to drop the "ignore" and "deprecated" options (and then rename "returncode" to "ignore"), but I was not sure if some people relied on the old implementation.